src/components/MultiColumnMeasures.tsx

```diff
@@ -1,163 +1,395 @@
+    () => multiColumnVariables[activeTab] || null,
+    [multiColumnVariables, activeTab]
+  );
 
-  const [termCards, setTermCards] = useState<TermCard[]>(
-    initializeTermCards({
-      columns,
-      terms,
-      assessmentToolColumns,
-      generateID,
-    })
+  // Memoized to prevent object lookup on every render
+  const currentState = useMemo(
+    () => (currentVariable ? variableStates[currentVariable.identifier] : null),
+    [currentVariable, variableStates]
   );
 
-  const itemsPerPage = 3;
-  const { currentPage, currentItems, totalPages, handlePaginationChange } = usePagination<TermCard>(
-    termCards,
-    itemsPerPage
+  // Memoized to prevent new array reference on every render from Zustand selector
+  const currentVariableColumns = useMemo(
+    () => (currentVariable ? getStandardizedVariableColumns(currentVariable) : []),
+    [currentVariable, getStandardizedVariableColumns]
   );
 
-  const allMappedColumns = getAllMappedColumns(termCards);
+  const currentTerms = currentState?.terms || [];
+
+  // Memoized to maintain stable reference for dependent hooks (useMemo and useCallback)
+  const currentTermCards = useMemo(() => currentState?.termCards || [], [currentState]);
+
+  const itemsPerPage = 3;
+  const { currentPage, totalPages, handlePaginationChange } =
+    usePagination<MultiColumnMeasuresTermCard>(currentTermCards, itemsPerPage);
+
+  // Memoized to prevent expensive array slicing operation on every render
+  const paginatedItems = useMemo(() => {
+    const startIndex = (currentPage - 1) * itemsPerPage;
+    return currentTermCards.slice(startIndex, startIndex + itemsPerPage);
+  }, [currentTermCards, currentPage, itemsPerPage]);
+
+  const variableAllMappedColumns = getAllMappedColumns(currentTermCards);
+
+  const loadTermsAndInitializeCards = useCallback(async () => {
+    if (multiColumnVariables.length === 0) {
+      setLoading(false);
+      return;
+    }
+
+    try {
+      const variablePromises = multiColumnVariables
+        .filter((variable) => !variableStates[variable.identifier])
+        .map(async (variable) => {
+          try {
+            const terms = await getTermOptions(variable);
+            const variableColumns = getStandardizedVariableColumns(variable);
+            const termCards = initializeTermCards({
+              columns,
+              terms,
+              variableColumns,
+              generateID,
+            });
 
-  const handleAddNewCard = () => {
-    const newCard: TermCard = {
+            return {
+              identifier: variable.identifier,
+              state: { terms, termCards },
+            };
+          } catch (error) {
+            // show a notif error
+            return {
+              identifier: variable.identifier,
+              state: {
+                terms: [],
+                termCards: [{ id: generateID(), term: null, mappedColumns: [] }],
+              },
+            };
+          }
+        });
```

Comment:
I don't understand why this is async. Right now the files are locally available, so this should not be async. But even if the terms were in some remote location: why would I fetch this stuff every time the component mounts? Wouldn't it make more sense to handle this once on app startup? And then stick it in the store